internal/querycoord/index_checker.go

```diff
@@ -277,3 +313,37 @@ func (ic *IndexChecker) processHandoffAfterIndexDone() {
 		}
 	}
 }
+
+func (ic *IndexChecker) AcquireSegmentLock(ctx context.Context, taskID int64, segIDs []int64) error {
```

Comment:
IndexChecker.broker already has the acquireSegmentsReferLock interface, so there is no need to add the interface.